src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift

```diff
@@ -604,6 +604,17 @@ class HealthCertificateService {
 		}
 	}
 
+	func remove(decodingFailedHealthCertificate: DecodingFailedHealthCertificate) {
+		healthCertifiedPersons
+			.first {
+				$0.decodingFailedHealthCertificates.contains(decodingFailedHealthCertificate)
+			}?
+			.decodingFailedHealthCertificates
+			.removeAll {
+				$0 == decodingFailedHealthCertificate
+			}
```

Comment:
should we check not better for the uniqueIdentifier of the certificate?

Reply:
Equatable for `DecodingFailedHealthCertificate` compares the base45 string which is also unique among all certificates, as it literally is the whole certificate. And we can't compare the uniqueIdentifier in this case, as we would need to be able to decode the certificate first to get the identifier